test/query.test.js

```diff
@@ -2708,17 +2708,144 @@ describe('query', () => {
+      query.Create(collectionRef, { credentials: { password: 'sekrit' } })
+    )
+
+    // Login with permissions for newly created Collection
+    const newLogin = await client.query(
+      query.Login(newCollection['ref'], { password: 'sekrit' })
+    )
+
+    // Get the newly scoped client
+    const newClient = await util.getClient({ secret: newLogin['secret'] })
+
+    // Retrieve the current identity
+    const currentIdentity = await newClient.query(query.CurrentIdentity())
+    expect(currentIdentity).toBeTruthy()
+  })
+
+  test('has_current_identity returns false without identity present', async () => {
+    const currentIdentity = await adminClient.query(query.HasCurrentIdentity())
+    expect(currentIdentity).toBeFalsy()
+  })
+
+  test('current_token returns object with AccessProvider', async () => {
+    const roleOneName = util.randomString('role_one_')
+    const issuerName = util.randomString('issuer_')
+    const providerName = util.randomString('provider_')
+    const jwksUri = util.randomString()
+    const fullUri = `https://${jwksUri}.auth0.com`
+
+    // Create Role
+    await adminClient.query(
+      query.CreateRole({
+        name: roleOneName,
+        privileges: [
+          {
+            resource: query.Databases(),
+            actions: { read: true },
+          },
+        ],
+      })
+    )
+
+    // Create AccessProvider
+    await adminClient.query(
+      query.CreateAccessProvider({
+        name: providerName,
+        issuer: issuerName,
+        jwks_uri: fullUri,
+        membership: [query.Role(roleOneName)],
+      })
+    )
+
+    // Retrieve Ref
+    const currentToken = await adminClient.query(query.CurrentToken())
+    expect(currentToken).toBeInstanceOf(Object)
+  })
+
+  test('current_token fails when unauthenticated', async () => {
+    try {
+      await adminClient.query(query.CurrentToken())
+    } catch (err) {
+      expect(err).toBeInstanceOf(errors.BadRequest)
+    }
+  })
+
+  test('has_current_token returns true when token present', async () => {
+    const roleOneName = util.randomString('role_one_')
+    const issuerName = util.randomString('issuer_')
+    const providerName = util.randomString('provider_')
+    const jwksUri = util.randomString()
+    const fullUri = `https://${jwksUri}.auth0.com`
 
-  // TODO Finish test after Core work is done
-  test.skip('current_token', () => {})
+    // Create Role
+    await adminClient.query(
+      query.CreateRole({
+        name: roleOneName,
+        privileges: [
+          {
+            resource: query.Databases(),
+            actions: { read: true },
+          },
+        ],
+      })
+    )
+
+    // Create AccessProvider
+    await adminClient.query(
+      query.CreateAccessProvider({
+        name: providerName,
+        issuer: issuerName,
+        jwks_uri: fullUri,
+        membership: [query.Role(roleOneName)],
+      })
+    )
+
+    const hasCurrentToken = adminClient.query(query.HasCurrentToken())
```

Comment:
same here, if you are using `adminClient` you are using a fauna key, not JWT token, so access provider is useless